### PR TITLE
docs: Update pieces of the documentation

### PR DESCRIPTION
**Pull request by jjerphan**
Hi,

Thank you for developing ArcticDB!

Here are some suggestions to improve the documentation based on my experience compiling, installing, and using ArcticDB.

I do not know if the `<figure markdown>` is needed or not (at least it seems that their path is wrong).

### CODE REVIEW

File: docs/mkdocs/docs/technical/architecture.md [hunk 1]
```diff
@@ -8,18 +8,12 @@ Users interact with the C++ storage engine, and therefore the storage itself, vi
 
 Please note that there is **no required server** component.
 
-<figure markdown>
-  ![Image title](../../images/ArcticArchitecture.png)
-  <figcaption>ArcticDB Architecture</figcaption>
-</figure>
+![ArcticDB Architecture](../images/ArcticArchitecture.png)
```
Comment (qc00) [inline docs/mkdocs/docs/technical/architecture.md:11]: I agree there isn't much value add from `<figure>` at the moment, especially given the captions are never changed.

@mehertz What was the original intention?

File: docs/mkdocs/docs/technical/contributing.md [hunk 2]
```diff
@@ -34,7 +34,10 @@ $MY_PYTHON -m pip install -U pip setuptools wheel grpcio-tools
 ARCTIC_CMAKE_PRESET=skip $MY_PYTHON setup.py develop
 # Change the below Python_EXECUTABLE value to build against a different Python version
 cmake -DPython_EXECUTABLE=$MY_PYTHON -DTEST=off --preset linux-debug cpp
-cmake --build --preset linux-debug cpp
```
Comment (qc00) [inline docs/mkdocs/docs/technical/contributing.md:37]: This command is indeed using argument combinations not officially supported by `cmake`. Did it not work for you?

The more portable command is

```bash
cd cpp
cmake --build --preset linux-debug
```

Comment (jjerphan) [reply docs/mkdocs/docs/technical/contributing.md:37]: Indeed `cmake --build --preset linux-debug cpp` was not working on my machine.

```
cmake version 3.26.0
```

Do the proposed command (i.e. `cmake -S cpp --preset linux-release`) work for you or would you like to use the ones you have proposed?

Comment (qc00) [reply docs/mkdocs/docs/technical/contributing.md:37]: `cmake` without the `--build` parameter does the configure step. We want to do the build step at this point.

**Review by qc00 — APPROVED** [review]

**Pull request merged**